src/main/kotlin/io/github/mojira/arisa/infrastructure/jira/Functions.kt

```diff
@@ -18,142 +23,178 @@ import java.net.URI
+    JiraClient(url, TokenCredentials(username, password))
+
 fun getIssue(jiraClient: JiraClient, key: String) = runBlocking {
     Either.catch {
         jiraClient.getIssue(key, "*all", "changelog")
     }
 }
+
 const val MILLI_FOR_FORMAT = 123L
 
-fun updateCHK(issue: Issue, chkField: String): Either<Throwable, Unit> = runBlocking {
-    Either.catch {
-        issue
-            .transition()
-            .field(
-                chkField,
-                Instant
-                    .now()
-                    .with(ChronoField.NANO_OF_SECOND, 0)
-                    .with(ChronoField.MILLI_OF_SECOND, MILLI_FOR_FORMAT)
-                    .toString()
-                    .replace("Z", "-0000")
-            )
-            .execute("Update Issue")
-    }
+fun updateCHK(context: IssueUpdateContext, chkField: String) {
+    context.hasUpdates = true
+    context.update.field(
+        chkField,
+        Instant
+            .now()
+            .with(ChronoField.NANO_OF_SECOND, 0)
+            .with(ChronoField.MILLI_OF_SECOND, MILLI_FOR_FORMAT)
+            .toString()
+            .replace("Z", "-0000")
+    )
 }
 
-fun updateConfirmation(issue: Issue, confirmationField: String, value: String) = runBlocking {
-    Either.catch {
-        val jsonValue = JSONObject()
-        jsonValue["value"] = value
+fun updateConfirmation(context: IssueUpdateContext, confirmationField: String, value: String) {
+    val jsonValue = JSONObject()
+    jsonValue["value"] = value
 
-        issue
-            .transition()
-            .field(confirmationField, jsonValue)
-            .execute("Update Issue")
-    }
+    context.hasUpdates = true
+    context.update.field(confirmationField, jsonValue)
 }
 
-fun updateLinked(issue: Issue, linkedField: String, value: Double) = runBlocking {
-    Either.catch {
-        issue
-            .transition()
-            .field(linkedField, value)
-            .execute("Update Issue")
-    }
+fun updateLinked(context: IssueUpdateContext, linkedField: String, value: Double) {
+    context.hasUpdates = true
+    context.update.field(linkedField, value)
 }
 
-fun deleteAttachment(jiraClient: JiraClient, attachment: Attachment): Either<Throwable, Unit> = runBlocking {
-    Either.catch {
-        jiraClient.restClient.delete(URI(attachment.self))
-        Unit
-    }
+fun reopen(context: IssueUpdateContext) {
+    context.transitionName = "Reopen Issue"
 }
 
-fun connectToJira(username: String, password: String, url: String) =
-    JiraClient(url, TokenCredentials(username, password))
+fun resolveAs(context: IssueUpdateContext, resolution: String) {
+    val resolutionJson = JSONObject()
+    resolutionJson["name"] = resolution
 
-fun reopenIssue(issue: Issue) = runBlocking {
-    Either.catch {
-        issue.transition().execute("Reopen Issue")
-    }
+    context.resolve.field(Field.RESOLUTION, resolutionJson)
+    context.transitionName = "Resolve Issue"
 }
```

Comment:
You are executing everything from all modules after a run, right?
What happens if there are two conflicting resolutions (e.g. Duplicate from CrashModule and Invalid from PiracyModule)?

Reply:
The one that is changed later will take effects, I believe

Reply:
Looked at the lib code, seems to be a regular map, so the last put should indeed be the one that takes effect.

We might want to prefer Dupe over everything else, as that's usually the most helpful, but since it probably still does the linking and post all the comments, it should really be fine.
We can always change it